### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by StasOkhrym**

### CODE REVIEW

File: cinema/views.py [hunk 1]
```diff
@@ -1,45 +1,124 @@
...
-        return Response(serializer.data, status=status.HTTP_400_BAD_REQUEST)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
-@api_view(["GET", "PUT", "DELETE"])
-def movie_detail(request, pk):
-    movie = get_object_or_404(Movie, pk=pk)
+class GenreDetail(APIView):
+    def get_object(self, pk: int) -> Genre:
+        try:
```
Comment (rostyak) [inline cinema/views.py:34]: you can use get_object_or_404(Genre, pk=pk), less code) but your variant is not a mistake

File: cinema/views.py [hunk 2]
```diff
@@ -1,45 +1,124 @@
...
+
+        return Response(serializer.data)
+
+    def patch(self, request: Request, pk: int) -> Response:
+        genre = self.get_object(pk)
+        serializer = GenreSerializer(genre, data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
```
Comment (rostyak) [inline cinema/views.py:51]: status?

File: cinema/views.py [hunk 3]
```diff
@@ -1,45 +1,124 @@
...
+    def put(self, request: Request, pk: int) -> Response:
+        genre = self.get_object(pk)
+        serializer = GenreSerializer(genre, data=request.data)
 
-    if request.method == "PUT":
-        serializer = MovieSerializer(movie, data=request.data)
         if serializer.is_valid():
             serializer.save()
-            return Response(serializer.data, status=status.HTTP_200_OK)
+            return Response(serializer.data)
```
Comment (rostyak) [inline cinema/views.py:61]: status?

File: cinema/views.py [hunk 4]
```diff
@@ -1,45 +1,124 @@
...
+        genre = self.get_object(pk)
+        serializer = GenreSerializer(genre, data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def put(self, request: Request, pk: int) -> Response:
```
Comment (rostyak) [inline cinema/views.py:55]: Do your methods "put" and "patch" do the same?

**Review by rostyak — CHANGES_REQUESTED** [review]

File: cinema/urls.py [hunk 5]
```diff
@@ -1,10 +1,44 @@
...
+    path("genres/<int:pk>/", GenreDetail.as_view(), name="genres-detail"),
+    path("actors/", ActorList.as_view(), name="genres-list"),
+    path("actors/<int:pk>/", ActorDetail.as_view(), name="genres-detail"),
+    path("cinema_halls/", cinema_hall_list, name="cinema-hall-list"),
+    path(
+        "cinema_halls/<int:pk>/",
+        cinema_hall_detail,
+        name="cinema-hall-detail"
+    ),
+    path("", include(router.urls)),
```
[the remark this answers is not in the document]
Comment (StasOkhrym) [reply cinema/urls.py:41]: https://www.django-rest-framework.org/api-guide/routers/#usage
routers do it automatically

**Review by rostyak — APPROVED** [review]

**Review by Nattalli — APPROVED** [review]

**Pull request closed without merging**